agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -2135,18 +2156,35 @@ bool backhaul_manager::handle_client_capability_query(ieee1905_1::CmduMessageRx
         return false;
     }
 
-    //if it is an error scenario, set Success status to 0x01 = Failure and do nothing after it.
+    // if it is an error scenario, set Success status to 0x01 = Failure and do nothing after it.
     if (associated_client_found) {
         client_capability_report_tlv->result_code() = wfa_map::tlvClientCapabilityReport::SUCCESS;
         LOG(DEBUG) << "Result Code: SUCCESS";
-        //TODO: Add frame body of the most recently received (Re)Association Request frame from this client
+        // Add frame body of the most recently received (Re)Association Request frame from this client
+        // auto associated_clients = m_radio_info_map[s.radio_mac].associated_clients_map[s.bssid];
+        // auto sAssociatedClients = associated_clients[client_info_tlv_r->client_mac()];
+        auto assoc_    = s.assoc_req;
+        auto assoc_req = std::string(assoc_, assoc_ + s.asso_len);
+
+        auto assoc_frame_body = string_utils::hex_to_char_string(assoc_req);
+
+        LOG(DEBUG) << "******************************************************************";
+        LOG(DEBUG) << "          sAssociatedClients.asso_len / 2 = " << s.asso_len / 2;
+        LOG(DEBUG) << "          assoc_req.len = " << assoc_req.length() / 2;
+        LOG(DEBUG) << "          assoc_frame_body.len = " << assoc_frame_body.length();
+
+        LOG(DEBUG) << "******************************************************************";
+        client_capability_report_tlv->alloc_association_frame(s.asso_len / 2);
+        std::copy(&assoc_frame_body[0], &assoc_frame_body[0] + s.asso_len / 2,
+                  client_capability_report_tlv->association_frame());
+
```

Comment:
This whole thing is wrong, no string conversion is needed here.

```suggestion
        client_capability_report_tlv->alloc_association_frame(s.asso_len);
        std::copy_n(s.assoc_req, s.asso_len, client_capability_report_tlv->association_frame());
```

Reply:
sorry, the branch was not updated, that's why I named the commit message WIP :)
I've pushed now the side branch, and I'll go over all your comments @arnout, thanks.